test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -96,37 +104,62 @@
       ;; Non-UTF-8 byte array
       (is (thrown? Exception (parse-command {:body (.getBytes "{\"command\": \"foo\", \"version\": 2, \"payload\": \"meh\"}" "UTF-16")}))))))
 
+(defn unroll-old-command [{:keys [command version payload]}]
+  [command
+   version
+   (or (:certname payload)
+       (:name payload))
+   payload])
+
 (defmacro test-msg-handler*
-  [command publish-var discard-var db & body]
-  `(let [log-output#     (atom [])
-         publish#        (call-counter)
-         discard-dir#    (fs/temp-dir "test-msg-handler")
-         handle-message# (mql/create-message-handler
-                          publish# discard-dir# #(process-command! % ~db))
-         msg#            {:headers {:id "foo-id-1"
-                                    :received (tfmt/unparse (tfmt/formatters :date-time) (now))}
-                          :body (json/generate-string ~command)}]
-     (try
-       (binding [*logger-factory* (atom-logger log-output#)]
-         (handle-message# msg#))
-       (let [~publish-var publish#
-             ~discard-var discard-dir#]
-         ~@body
-         ;; Uncommenting this line can be very useful for debugging
-         ;; (println @log-output#)
-         )
-       (finally
-         (fs/delete-dir discard-dir#)))))
+  [command delay-var discard-var db & body]
+  `(tqueue/with-stockpile q#
```

Comment:
I tried to keep the changes to this test code to a minimum. I think that it's simpler to test these pieces integrated now, it might not be as necessary to have all the moving parts. The existing tests work though, just switching the calls like I have here